packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -716,8 +716,48 @@ describe("BnsConnection", () => {
       connection.disconnect();
     });
 
+    it("can search for transactions by hash", async () => {
+      pendingWithoutBnsd();
+      const connection = await BnsConnection.establish(bnsdTendermintUrl);
+      const chainId = await connection.chainId();
+
+      const { profile, mainWalletId, faucet } = await userProfileWithFaucet();
+      const faucetAddress = keyToAddress(faucet.pubkey);
+
+      const memo = `Payment ${Math.random()}`;
+      const sendTx: SendTx = {
+        kind: TransactionKind.Send,
+        chainId: chainId,
+        signer: faucet.pubkey,
+        recipient: await randomBnsAddress(),
+        memo: memo,
+        amount: {
+          quantity: "1000000001",
+          fractionalDigits: 9,
+          tokenTicker: cash,
+        },
+      };
+
+      const nonce = await getNonce(connection, faucetAddress);
+      const signed = await profile.signTransaction(mainWalletId, faucet, sendTx, bnsCodec, nonce);
+      const txBytes = bnsCodec.bytesToPost(signed);
+      const response = await connection.postTx(txBytes);
+      const transactionIdToSearch = response.data.txid;
+
+      await sleep(50); // Tendermint needs some time to update search index
```

Comment:
Later we can move this to a test config parameter so we can switch between environments.

Reply:
Jap! It took us a while to understand the issue here. Now we can deduplicate code and comment.

It turns out that some ms sleep are even needed on a fast local development machine. Those 50 ms work for local and Travis. For production networks, a longer sleep may be necessary. However, the tests only run against local chains. Let's see